src/azure-cli-core/azure/cli/core/aaz/_field_value.py

```diff
@@ -444,3 +445,113 @@ def to_serialized_data(self, processor=None, keep_undefined_in_list=False,  # py
+                    if v == AAZUndefined:
+                        continue
+
+                    if field_schema._serialized_name:
+                        name = field_schema._serialized_name
+
+                    if name in {
+                        self.M_SYSTEM_ASSIGNED, self.M_USER_ASSIGNED,
+                        self.S_SYSTEM_ASSIGNED, self.S_USER_ASSIGNED,
+                    }:
+                        data[name] = v
+                        if action := field_schema._flags.get("action", None):  # subcommand has action
+                            data["action"] = action
+
+                    else:
+                        result[name] = v
+
+        result = self.build_identity(data, result)
+
+        if not result:
+            result = {"type": "None"}  # empty identity
+
+        if not result and self._is_patch:
+            result = AAZUndefined
+
+        if processor:
+            result = processor(self._schema, result)
+
+        return result
+
+    def build_identity(self, data, result):
+        if not data:
+            return result
+
+        if action := data.get("action", None):  # subcommand
+            identities = set(result.pop("userAssignedIdentities", {}).keys())
+            has_system_identity = "systemassigned" in result.pop("type", "").lower()
+
+            user_assigned = data.get(self.S_USER_ASSIGNED, None)
+            system_assigned = data.get(self.S_SYSTEM_ASSIGNED, None)
+
+            if action == "remove":
+                if user_assigned is not None:
+                    if len(user_assigned) > 1:  # remove each
+                        identities -= set(user_assigned)
+
+                    else:  # remove all
+                        identities = {}
+
+                if identities:
+                    result["userAssignedIdentities"] = {k: {} for k in identities}
+                    if system_assigned or not has_system_identity:
+                        result["type"] = "UserAssigned"
+
+                    else:
+                        result["type"] = "SystemAssigned, UserAssigned"
+
+                elif not system_assigned and has_system_identity:
+                    result["type"] = "SystemAssigned"
+
+            elif action == "assign":
+                if user_assigned:
+                    identities |= set(user_assigned)
+
+                if identities:
+                    result["userAssignedIdentities"] = {k: {} for k in identities}
+                    if system_assigned or has_system_identity:
+                        result["type"] = "SystemAssigned, UserAssigned"
+
+                    else:
+                        result["type"] = "UserAssigned"
+
+                elif system_assigned or has_system_identity:
+                    result["type"] = "SystemAssigned"
+
+        else:  # main command
+            mi_user_assigned = data.get(self.M_USER_ASSIGNED, None)
+            mi_system_assigned = data.get(self.M_SYSTEM_ASSIGNED, None)
```

Comment:
Will the update command support mi_user_assigned and mi_system_assigned?

Reply:
If the update command supported, then the mi_user_assigned or mu_system_asssigned could be undefined and you should think about the values from get as well to determine the final type.

Reply:
update command won't support that,
https://github.com/necusjz/aaz-dev-tools/blob/96f0f5aa2e415d94c67986d9bb595c77658a0eef/src/aaz_dev/command/model/configuration/_arg_builder.py#L196